src/gtk/textctrl.cpp

```diff
@@ -560,6 +560,29 @@ au_insert_text_callback(GtkTextBuffer *buffer,
         wxGtkTextApplyTagsFromAttr(win->GetHandle(), buffer, win->GetDefaultStyle(),
                                    &start, end);
     }
+    auto maxlen = win->GTKGetMaxLength();
+    auto count = gtk_text_buffer_get_char_count( buffer );
+    if( maxlen > 0 && count > maxlen )
+    {
+        GtkTextIter offset;
+        int trim_len = count - maxlen;
+        gtk_text_buffer_get_iter_at_offset( buffer, &offset, gtk_text_iter_get_offset( end ) - trim_len );
+        // This function is connected using g_signal_connect_after() call, therefore
+        // direct buffer modification is required.
+        // Using g_signal_connect() doesn't work as the text is still being entered
+        // probably because the signal is documented as "Run Last". For exactly same
+        // reason, ("Run Last") it won't work in GTKOnInsertText() as it called from
+        // the handler that does not connected after
+        gtk_text_buffer_delete( buffer, &offset, end );
+#if GTK_CHECK_VERSION( 3, 2, 0 )
+        if( wx_is_at_least_gtk3(2) )
+            gtk_text_iter_assign( end, &offset );
+#endif
```

Comment:
What is this code for? `end` doesn't seem to be used any more, so why do we bother changing it? And, of course, if we do need to change it, what's wrong with just writing `end = offset`??

```suggestion
```

Reply:
@vadz , I tried to get rid of this and didn't see any bad side effects. Lets's see how it will work with other GTK+ version...